### PR TITLE
fix wrong link in user and organization profile when using relative url

### LINKED ISSUE #28436: Relative URLs not working in profile page README
### Description

In a profile repository README, like in any repository markdown file, relative URLs for links and images are converted when rendering.
But in a profile page, relative URLs in the README aren't converted, thus not working.

### Gitea Version

1.22.0+dev-420-g717d0f593

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

| Repository page | Profile page |
| --- | --- |
|  |  |

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Using https://try.gitea.io/KaKi87

### Database

_No response_

---

Thanks

### PR DESCRIPTION
**Pull request by katsusan**
fix #28436.
the doc https://docs.gitea.com/usage/profile-readme maybe also need to be updated to tell that
the main branch is necessary,which means the following three conditions should be satisfied:
- repo: **.profile**
- branch: **[default branch]**
- markdown: **README.md**

### CODE REVIEW

File: routers/web/user/profile.go [hunk 1]
```diff
@@ -233,10 +234,20 @@ func prepareUserProfileTabData(ctx *context.Context, showPrivate bool, profileGi
 		if bytes, err := profileReadme.GetBlobContent(setting.UI.MaxDisplayFileSize); err != nil {
 			log.Error("failed to GetBlobContent: %v", err)
 		} else {
+			// When user opens someone's profile page, the router doesn't give the repo context, which makes it difficult to
+			// render the full link of media elements.
+			// the media link should be in the format of /[user]/[repoName]/src/branch/[branchName],
+			// 	Eg. /Tom/.profile/src/branch/main
+			// The branch shown on the profile page should be the 'main' branch, this need to be in sync with doc, see:
+			//	https://docs.gitea.com/usage/profile-readme
+			const profileBranch = "main"
```
Comment (lunny) [inline routers/web/user/profile.go]: It's should be `repo.DefaultBranch`?

File: routers/web/user/profile.go [hunk 2]
```diff
@@ -233,10 +234,20 @@ func prepareUserProfileTabData(ctx *context.Context, showPrivate bool, profileGi
...
+			repoName := strings.TrimSuffix(path.Base(profileGitRepo.Path), ".git")
```
Comment (wxiaoguang) [inline routers/web/user/profile.go]: At least 2 problems:

1. `path` is not right to process the "git repo path", which is a file system path
2. this assumption is not right. "git repo path" doesn't always mean "reop name"

File: routers/web/user/profile.go [hunk 3]
```diff
@@ -233,10 +234,20 @@ func prepareUserProfileTabData(ctx *context.Context, showPrivate bool, profileGi
...
+			prefix := fmt.Sprintf("/%s/%s/src/branch/%s", ctx.ContextUser.Name, repoName, profileBranch)
```
Comment (wxiaoguang) [inline routers/web/user/profile.go]: When using uncertain contents like "profileBranch" (which should come from DefaultBranch), it should be escaped before putting it in a URL.

File: routers/web/user/profile.go (hunk 1 above)
Comment (katsusan) [reply routers/web/user/profile.go]: Done. I'm not sure if the doc should be updated.

File: routers/web/user/profile.go (hunk 2 above)
Comment (katsusan) [reply routers/web/user/profile.go]: > path is not right to process the "git repo path", which is a file system path
> this assumption is not right. "git repo path" doesn't always mean "reop name"
About branch 

Done. Now using repo_model.Repository's metadata for repo path.

File: routers/web/user/profile.go (hunk 3 above)
Comment (katsusan) [reply routers/web/user/profile.go]: Done

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**